t/21-needles.t

```diff
@@ -168,4 +180,69 @@ subtest 'handling relative paths in update_needle' => sub {
     };
 };
 
+subtest 'controller->_determine_needles_dir_for_job' => sub {
+    my $controller = OpenQA::WebAPI::Controller::Step->new;
+    $controller->app($t->app);
+    $settings{CASEDIR} = 'https://something#fragment';
+    my $sandbox = tempdir(CLEANUP => 1);
+    my $job = $schema->resultset('Jobs')->create_from_settings(\%settings);
+    $job->update({result_dir => $sandbox});
+
+    subtest 'checkout_needles_sha = no' => sub {
+        $controller->app->config->{'scm git'} = {checkout_needles_sha => 'no'};
+        my ($needle_dirs, $needles_var) = $controller->_determine_needles_dir_for_job($job);
+        is($needle_dirs, undef, 'needles directory determined');
+        is($needles_var, undef, 'needles var determined');
+    };
+
+    subtest 'checkout_needles_sha = yes' => sub {
+        $controller->app->config->{'scm git'} = {checkout_needles_sha => 'yes'};
+        my ($needle_dirs, $needles_var) = $controller->_determine_needles_dir_for_job($job);
+        like($needle_dirs, qr|t/data/openqa/share/tests/fedora/needles$|, 'needles dirs is defined');
+        is($needles_var->value, 'https://something#fragment', 'needles var is defined');
+    };
+};
+
+subtest 'controller->_create_tmpdir_for_needles_refspec' => sub {
+    my $tempdir = setup_fullstack_temp_dir('21-needles');
+    my $testresult_dir = $tempdir . '21-needles/openqa/testresults/00000';
+    path($testresult_dir)->make_path;
+    my $vars_json = path($testresult_dir, 'vars.json');
+    my $vars_json_content = '{"NEEDLES_GIT_HASH": "0123456789abcdef0123456789abcdef01234567"}';
+    $vars_json->spew($vars_json_content);
+
+    my $controller = OpenQA::WebAPI::Controller::Step->new;
+    $controller->app($t->app);
+    $settings{CASEDIR} = 'https://something#fragment';
+    my $job = $schema->resultset('Jobs')->create_from_settings(\%settings);
+    path($job->needle_dir)->make_path;
+    $controller->app->config->{'scm git'} = {checkout_needles_sha => 'yes'};
+
+    my $needle_refs = $controller->_create_tmpdir_for_needles_refspec($job);
+
+    is($needle_refs, '0123456789abcdef0123456789abcdef01234567', 'needles refs is defined with a fragment');
+
+    # TODO: missing line 164
+};
+
+subtest 'needledir set by jsonfile_in_temp_dir' => sub {
+    # Data
+    my $cwd = Cwd::getcwd;
```

Comment:
What is the plan here?
1. We have `$FindBin::Bin` for the directory the test is in. Relying on cwd is usually not needed
2. But we agreed that we don't want to mess with the git directory and instead put stuff into a tempdir

Reply:
Fixed